quinn/src/runtime/mod.rs

```diff
@@ -0,0 +1,42 @@
+#[cfg(feature = "runtime-tokio")]
+mod tokio_runtime;
+#[cfg(feature = "runtime-tokio")]
+pub use tokio_runtime::*;
+
+#[cfg(feature = "runtime-async-std")]
+mod async_std_runtime;
+#[cfg(feature = "runtime-async-std")]
+pub use async_std_runtime::*;
+
+use std::fmt::Debug;
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use std::time::Instant;
+
+pub trait AsyncTimer: Send + Debug {
+    fn reset(self: Pin<&mut Self>, i: Instant);
+    fn poll(self: Pin<&mut Self>, cx: &mut Context) -> Poll<()>;
+}
+
+pub trait Runtime: udp::runtime::Runtime {
+    fn new_timer(&self, i: Instant) -> Pin<Box<dyn AsyncTimer>>;
+    fn spawn(&self, future: Pin<Box<dyn Future<Output = ()> + Send>>);
+}
+
+pub fn make_runtime() -> Box<dyn Runtime> {
```

Comment:
I'd prefer not having this automatic selection, instead requiring callers to pass in an `impl Runtime` explicitly.

Reply:
You'll have to discuss this with @Ralith, who requested to reduce the changes to the user-facing API.

Reply:
I think it will be very rare for this not to do the expected thing, and in those cases `Endpoint::new` can be used. I'd rather not complicate every entry point with a detail that most users aren't interested in customizing.